designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,139 @@
+
+## Bulk actions in toolbar
+The bulk selector will be the first component in the toolbar on pages where bulk actions are available.
+![toolbar](img/toolbar.png)
+
+Users can click the checkbox in the bulk selector to select all items in the list. If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read ‘[#] selected’, informing users that all known items in the list are selected. Clicking the bulk selector checkbox again will deselect all items.
+![bulk-selector-states](img/bulk-selector-states.png)
+
+Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a ‘-‘ to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. ‘2 selected’). Clicking the bulk selector checkbox will deselect all items.
+![add-labels](img/add-labels.png)
+
+## Behavior
+If no items are selected, the bulk actions dropdown will be enabled, shown as a basic dropdown, but the actions within will be disabled. A tooltip will explain to users that they must select resources in order to take actions.
+![none-checked](img/none-checked.png)
+
+When items are selected, the bulk actions basic dropdown will turn into a primary dropdown, and actions within it will be enabled. 
+![add-labels](img/add-labels.png)
+
+### Item limits
+Depending on the use case, there may be scenarios where users are limited to selecting a certain number of items, assuming there is a large total in a dataset. As a general rule of thumb, the max number of items that can be selected at once will be set at 200. 
+![max-number-200](img/max-number-200.png)
+
+Once the user hits that limit, they will not be able to select more items and all other checkboxes will be disabled. 
+![max-number-error](img/max-number-error.png)
+
+### Permission limits
+Depending on the use case, there may be scenarios where users are limited to a certain set of actions to perform on items. If the user does not have permission to perform a specific action on **any** of the selected items, the action will still be enabled in the dropdown. Once they select the action, the modal will contain a failure alert, explaining that the action cannot be taken.
+![no-permissions](img/no-permissions.png)
+
+If the user has permission to perform a specific action on **some** of the selected items, the action will be carried out for all of the items that it can be. A failure alert will be included within the modal to list the items that failed to complete.
+![some-permissions](img/some-permissions.png)
+
+
+### Infinite scrolling
+If a list view does not contain pagination and includes infinite scrolling, when the user uses the bulk selector to select all items, only the items within view would be selected. 
```

Comment:
Is it important to notify the user that this is the behavior? Wondering if some users would assume it's actually all, not just all loaded.

Reply:
The count of all items will (hopefully if all toolbars get updated) be displayed as well as the count of selected items within the bulk selector - do you think that's enough for them to figure out that it's only the selected items in view?

Reply:
I get why this interaction would need to be this way but it seems slightly unfortunate the user has to scroll down as far as they care to (maybe really far) just to scroll all the way back up to select all.

I was curious since this seems like a sort of "classic UX" type question and saw this thread: https://ux.stackexchange.com/questions/113911/infinite-scroll-and-select-all
At least one person mentioned not using select all at all with infinite scroll, and honestly I could see the merit in the decision.

Reply:
@itsptk Another solution could be to make the toolbar sticky on our tables - PF offers that now. Thoughts?

Reply:
After thinking about this some more I am going to update my opinion to say that when there is infinite scrolling, select all should select every item in the list (loaded or not) or we should probably not use infinite scrolling or not have a select all. Infinite scrolling seems like an implementation detail and not something the user should otherwise be concerned with, how many items are loaded to be selected, etc.

I suspect this could probably play into why well known apps (gmail) use pagination and not infinite scroll, to avoid the selection issues. The pagination mechanism of 'select everything on the page' or 'select everything entirely' is operating on the page mechanism that is more directly exposed to the user ( items 1-50, next page >), which makes more sense to me.

Reply:
Thought about this for a bit and decided to update it based on the newest updates to bulk selection demos in PF! Selecting the bulk selector checkbox will "select all items (200 items max)" and there will be an additional option to "select page (## items)" to only select items in view 
@bmignano @itsptk thank you for your input and ideas here!!!